src/elements/content-sidebar/SidebarPanels.js

```diff
@@ -85,87 +83,97 @@ const SidebarPanels = ({
+                    exact
+                    path={`/${SIDEBAR_VIEW_ACTIVITY}`}
+                    render={() => (
                         <LoadableActivitySidebar
                             currentUser={currentUser}
                             file={file}
                             onVersionHistoryClick={onVersionHistoryClick}
                             startMarkName={MARK_NAME_JS_LOADING_ACTIVITY}
                             {...activitySidebarProps}
                         />
-                    )
-                }
-            />
-            <SidebarRoute
-                enabled={hasDetails}
-                exact
-                path={`/${SIDEBAR_VIEW_DETAILS}`}
-                pathFallback={`/${SIDEBAR_VIEW_METADATA}`}
-                render={() =>
-                    isOpen && (
+                    )}
+                />
+            )}
+            {hasDetails && (
+                <Route
+                    exact
+                    path={`/${SIDEBAR_VIEW_DETAILS}`}
+                    render={() => (
                         <LoadableDetailsSidebar
                             fileId={fileId}
                             key={fileId}
                             onVersionHistoryClick={onVersionHistoryClick}
                             startMarkName={MARK_NAME_JS_LOADING_DETAILS}
                             {...detailsSidebarProps}
                         />
-                    )
-                }
-            />
-            <SidebarRoute
-                enabled={hasMetadata}
-                path={`/${SIDEBAR_VIEW_METADATA}`}
-                pathFallback={`/${SIDEBAR_VIEW_SKILLS}`}
-                render={() =>
-                    isOpen && (
+                    )}
+                />
+            )}
+            {hasMetadata && (
+                <Route
+                    exact
+                    path={`/${SIDEBAR_VIEW_METADATA}`}
+                    render={() => (
                         <LoadableMetadataSidebar
                             fileId={fileId}
                             startMarkName={MARK_NAME_JS_LOADING_METADATA}
                             {...metadataSidebarProps}
                         />
-                    )
-                }
-            />
-            <SidebarRoute
-                enabled={hasVersions}
-                path="/:sidebar/versions/:versionId?"
-                pathFallback="/"
-                render={({ match }) => (
-                    <LoadableVersionsSidebar
-                        fileId={fileId}
-                        key={fileId}
-                        onVersionChange={onVersionChange}
-                        parentName={match.params.sidebar}
-                        versionId={match.params.versionId}
-                    />
-                )}
+                    )}
+                />
+            )}
+            {hasVersions && (
+                <Route
+                    path="/:sidebar/versions/:versionId?"
```

Comment:
You mentioned moving to a model where a version is always selected. Do we we need to remove this optionality?

Reply:
Not yet, unfortunately. We don't always have access to the current version id in the host application at the moment, so there will be some cases where it's empty. I'd like to improve that in the near future, though.